accounts/backends.py

```diff
@@ -35,16 +37,23 @@ def authenticate(self, request, remote_user, shibboleth_attributes):
         User = get_user_model()
         username = self.clean_username(remote_user)
         user, created = User.objects.get_or_create(username=username)
+
+        # Add inital attributes on first log in
         if created:
             user.set_unusable_password()
             if isinstance(shibboleth_attributes, Mapping):
                 for key, value in shibboleth_attributes.items():
-                    if value:
-                        setattr(user, key, value)
-                    else:
-                        setattr(user, key, self.searchPennDirectory(username, key))
-
+                    if key != 'affiliation':
+                        setattr(user, key, value if value else self.searchPennDirectory(username, key))
```

Comment:
Are we worried about latency issues for an external request to the Penn directory? What if the directory API is down, but shibboleth is still up? We should probably still accept logins in that case.

Reply:
Read the thread with @kirubarajan, but still think this is something to think about at least.

Reply:
That's a good question. Polling the directory only happens on a user's first login, so I don't think latency will be a real issue. We probably should add some error checking if the directory is down, but since we don't have access to it yet, I think we should leave it as is.